File: frontend/src/models/PreludeModel.ts
import type { PenaltyReason } from "./PenaltyReason.ts";

export interface PenaltyInformation {
    fine: number,
    due: Date,
    reason: PenaltyReason,
}

/**
 * General information required by multiple views.
 */
export interface PreludeModel {
    /** Minimal reservation time imposed by the server. */
    minReservationTime: number,

    /** Minimal reservation time imposed by the server. */
    maxReservationTime: number,

    /**
     * Information about the current user penalty. `null`
     * if no penalty is active.
     */
    penaltyInformation: PenaltyInformation | null
}

File: src/main/kotlin/models/PreludeModel.kt
package parkflex.models

import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import parkflex.db.PenaltyReason
import java.time.LocalDateTime


/**
 * General information required by multiple views.
 */
@Serializable
data class PreludeModel(
    /** Minimal reservation time imposed by the server. */
    val minReservationTime: Long,

    /** Minimal reservation time imposed by the server. */
    val maxReservationTime: Long,

    /**
     * Information about the current user penalty. `null`
     * if no penalty is active.
     */
    val penaltyInformation: PenaltyInformation?
) {
    @Serializable
    data class PenaltyInformation(
        val fine: Long,
        @Contextual
        val due: LocalDateTime,
        val reason: PenaltyReason
    )
}